react/components/SearchResultContainer.js

```diff
@@ -105,43 +107,82 @@ const SearchResultContainer = props => {
           ...prevResult,
           productSearch: {
             ...prevResult.productSearch,
-            products: [
-              ...prevResult.productSearch.products,
-              ...fetchMoreResult.productSearch.products,
-            ],
+            products: isForward
+              ? [
+                  ...prevResult.productSearch.products,
+                  ...fetchMoreResult.productSearch.products,
+                ]
+              : [
+                  ...fetchMoreResult.productSearch.products,
+                  ...prevResult.productSearch.products,
+                ],
           },
         }
       },
     })
   }
 
-  useFetchMoreOnStateChange(handleFetchMore, fetchMoreLoading)
+  const handleFetchMoreNext = () => {
+    const from = toRef.current + 1
+    const to = min(recordsFiltered, from + maxItemsPerPage) - 1
+    handleFetchMore(from, to, true)
+    toRef.current = to
+    pageRef.current = nextPageRef.current
+    nextPageRef.current += 1
+    setQuery({ page: pageRef.current }, { replace: true })
+  }
+
+  const handleFetchMorePrevious = () => {
+    const to = fromRef.current - 1
+    const from = max(0, to - maxItemsPerPage + 1)
+    handleFetchMore(from, to, false)
```

Comment:
using function parameters like this leads to code that's hard to read—it is not clear what "false" is doing here, for instance.

in these cases, I suggest making the `handleFetchMore` accept an object as argument, so you can use it like this:
```
handleFetchMore({
  from,
  to,
  isForward: false
})
```

this is one suggestion; another suggestion, however, is whenever a function receives a boolean as argument, think about splitting it into two different functions, and/or accept an enum as argument instead